samples/word/25-paragraph/insert-header-and-footer.yaml

```diff
@@ -1,57 +1,169 @@
+              "Start"
+            );
+            body.insertBreak(Word.BreakType.page, Word.InsertLocation.end);
+            body.insertParagraph(
+              "Video provides a powerful way to help you prove your point. When you click Online Video, you can paste in the embed code for the video you want to add. You can also type a keyword to search online for the video that best fits your document.",
+              "End"
+            );
+            body.insertBreak(Word.BreakType.page, Word.InsertLocation.end);
+            body.insertParagraph(
+              "To make your document look professionally produced, Word provides header, footer, cover page, and text box designs that complement each other. For example, you can add a matching Online cover page, header, and sidebar. Click Insert and then choose the Online elements you want from the different Online galleries.",
+              "End"
+            );
+
+            // Clear any headers and footers.
+            const section = context.document.sections.getFirst();
+
+            section.getHeader("Primary").clear();
+            section.getHeader("FirstPage").clear();
+            section.getHeader("EvenPages").clear();
+
+            section.getFooter("Primary").clear();
+            section.getFooter("FirstPage").clear();
+            section.getFooter("EvenPages").clear();
+          });
+        }
+
         /** Default helper for invoking an action and handling errors. */
         async function tryCatch(callback) {
           try {
             await callback();
           } catch (error) {
-              // Note: In a production add-in, you'd want to notify the user through your add-in's UI.
-              console.error(error);
+            // Note: In a production add-in, you'd want to notify the user through your add-in's UI.
+            console.error(error);
           }
         }
     language: typescript
 template:
     content: |-
         <section class="ms-font-m">
-            This sample inserts a header and footer in the document.
+            This sample inserts headers and footers in the document.
+        </section>
+
+        <section class="setup ms-font-m">
+            <h3>Set up</h3>
+            <button id="setup" class="ms-Button">
+                <span class="ms-Button-label">Add sample text</span>
+            </button>
         </section>
 
         <section class="samples ms-font-m">
             <h3>Try it out</h3>
+            <p>To see how the various headers and footers are displayed depending on your settings in the <b>Header & Footer</b> ribbon, refer to <a
+                    target="_blank" href="https://support.microsoft.com/office/490423a3-3e5b-4080-aff0-c429ab5d6708">Edit your
+                    existing headers and footers</a>.</p>
```

Comment:
Since the user needs to adjust these settings as they go to see the changes, consider restructuring this HTML task pane to be more procedural. Having steps between the button sections would help.